File: Team_3_Sign_Language_AI-Karishma-milestone3/frontend/src/types/courses.ts
// Mirrors backend/app/schemas/courses.py

export interface CourseItem {
  id: string;
  title: string;
  category: string;
  description: string;
  route: string | null;
  built: boolean;
  tracks_progress: boolean;
  // Null whenever progress isn't tracked or computable — never a
  // placeholder number for a course that isn't real yet.
  progress_percent: number | null;
  item_count: number | null;
  locked_reason: string | null;
}

export interface CourseCatalog {
  courses: CourseItem[];
}

File: Team_3_Sign_Language_AI-Karishma-milestone3/backend/app/schemas/courses.py
from typing import Optional

from pydantic import BaseModel


class CourseItem(BaseModel):
    id: str
    title: str
    category: str
    description: str
    route: Optional[str] = None
    built: bool
    tracks_progress: bool
    # Null whenever progress isn't tracked or isn't computable — never a
    # placeholder number for a course that doesn't exist yet.
    progress_percent: Optional[float] = None
    item_count: Optional[int] = None
    locked_reason: Optional[str] = None
    # Set only for conversational-fluency: the trained classifier's own
    # measured test accuracy, so the frontend can show real reliability
    # instead of implying this is as dependable as the alphabet
    # classifier. Null for every other course.
    model_test_accuracy: Optional[float] = None


class CourseCatalogResponse(BaseModel):
    courses: list[CourseItem]
